### DISCUSSION TITLE
Just a PDF viewer

### DISCUSSION THREAD

**Discussion by AndersonCardosoSilva-BR** [sumatrapdfreader/sumatrapdf#2381] [Ideas]
I need a report PDF viewer to my system program.
At src/previewer there is some code I presume just to view the PDF.
So I have a system program that make reports and I would like it had a native PDF viewer, I mean, it would shows its reports in the screen without calling any other program.
I think I could integrate a program written in C like some source somewhere in Sumatra to do that. I just need the minimum thing to generate a report viewer.
I need to compile this code with BCC55 or any BCC newer version to be compiled with my system.
Nowadays I use PDFCreator (a virtual printer driver) to transform my report to PDF and then open it with the standard PDF viewer installed. There is a way to program PDF generation but there is no way to program a PDF viewer using xHarbour, at least any I know. I use xHarbour v1.2.1.
Thanks in advance.

Anderson

**Comment by GitHubRulesOK** [comment]
Sumatra PDF license is clear its not a plugin to another application, thus in effect you may
- Call the unmodified compiled published **.exe**cutable, just the same as any other system application (many other apps use it this way, including Harbour) 
- OR publicly publish your fully Open Source application to honour the related AGPL, as SumatraPDF does
- here an application does both e.g. the source code was open (though difficult to find nowadays) and the pdf view using SumatraPDF.exe is reasonably faithful to allow annotation, even if a bit flakey at times but project was dropped before the bugs were ironed out beyond beta concept.
![image](https://user-images.githubusercontent.com/12754913/156802678-5b01b08e-d65a-4354-9a02-c09b140a12b0.png)
